### PR TITLE
[FSSDK-9032] enhancement: Add VUID instantiation to browser client promise dependencies

### PR DESCRIPTION
**Pull request by opti-jnguyen**
## Summary
- For browser clients, adds VUID instantiation promise to the client's onReady promise dependencies.

This helps ensure VUID successfully populates before the onReady promise returns for the client, preventing calls to other VUID-dependent functions from failing if VUID hasn't been initialized yet.

- Also adds `isBrowser` utility and includes various formatting cleanup.

## Test plan
- All existing tests should pass.

## Issues
- [FSSDK-9032](https://jira.sso.episerver.net/browse/FSSDK-9032)

### CODE REVIEW

File: packages/optimizely-sdk/lib/utils/fns/index.ts [hunk 1]
```diff
@@ -165,6 +163,18 @@ export function checkArrayEquality(arrayA: string[], arrayB: string[]): boolean
   return arrayA.length === arrayB.length && arrayA.every((item, index) => item === arrayB[index]);
 }
 
+/**
+ * Checks the current running context
+ * @returns {boolean} True if window object exists.
+ */
+export function isBrowser(): boolean {
```
Comment (mikechu-optimizely) [inline packages/optimizely-sdk/lib/utils/fns/index.ts:170]: Did you want to add to the fns unit tests?

**Review by mikechu-optimizely — APPROVED** [review]
LGTM unless we want to add some test coverage.

**Review by raju-opti — APPROVED** [review]
LGTM

**Comment by opti-jnguyen** [comment]
Note: Chatted with Raju regarding the implementation yesterday and will be considering a different implementation that will be part of the clean-up PR.

In the proposed re-implementation of this, the plan will be to always inject the odpManager's initPromise dependency regardless of Browser or Node but will not have Node's promise resolution dependent on the initialization of a VUID.

**Pull request merged**